File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    dic = {}
    
    for n in range(len(name)):
        dic[name[n]] = yearning[n]
        
    for i in range(len(photo)):
        total = 0

        for ph in photo[i]:
            if ph in dic:
                total += dic[ph]
                
        answer.append(total)
            
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<string> name, vector<int> yearning, vector<vector<string>> photo) {
    vector<int> answer;
    for(int i=0;i<photo.size();i++){
        int sum = 0;
        for(int j=0;j<photo[i].size();j++){
            auto it = find(name.begin(),name.end(),photo[i][j]);
            if(it != name.end()){
                sum += yearning[distance(name.begin(), it)];    
            }
            
        }
        answer.push_back(sum);
    }
    
    return answer;
}